### PR TITLE
feat: Collapsable sidebar in Safe App view

### LINKED ISSUE #866: [Epic] Safe Apps can only be viewed inside a limited frame
## Overview
Enhance the adaptability of the Safe interface to just wrap some of its elements to enable more screen space for the Safe App. This will make the Safe App to look as it's not used inside of a Safe until transaction time

Roadmap Epic:
https://www.notion.so/gnosis-safe/Expanding-the-Safe-App-frame-fe75278b7d22416bb4ea33171813e8ef

## Requirements
- [ ] Design a new frame view to get more space for Safe Apps
- [ ] Create interface elements to allow navigating again to the Safe
- [ ] Pop-up transaction modal still with Safe App open
- [ ] Show transaction queue with Safe App open if something is pending to be signed

## Designs
Need to sync with Lila

### PR DESCRIPTION
**Pull request by DaniSomoza**
## What it solves

Resolves #866

- [figma designs](https://www.figma.com/file/2T4n3zv8xbeeWWD3UHyqpf/App-frime-size?node-id=1%3A2&t=hY1csSZdWjNdBOXN-0) 

## How this PR fixes it

- Added a collapsable button bar in the Safe App view.
- Restored some css animations on collapse sidebar.
- After some conversations, removed the tooltip present in designs

## How to test it

- Open a safe app in desktop resolution
- Open & close the sidebar using the collapsable button bar in the left side.

## Screenshots

Captura de pantalla 2022-11-16 a las 14 12 59
Captura de pantalla 2022-11-16 a las 14 13 19

### CODE REVIEW

File: src/components/common/PageLayout/index.tsx [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
 import Footer from '../Footer'
 import { AppRoutes } from '@/config/routes'
+import { useSafeAppUrl } from '@/hooks/safe-apps/useSafeAppUrl'
+import { useMediaQuery } from '@mui/material'
 
 const PageLayout = ({ children }: { children: ReactElement }): ReactElement => {
   const router = useRouter()
-  const [isMobileDrawerOpen, setIsMobileDrawerOpen] = useState<boolean>(false)
-  const hideSidebar = router.pathname === AppRoutes.share.safeApp
+  const isMobile = useMediaQuery('(max-width: 900px)')
```
Comment (katspaugh) [inline src/components/common/PageLayout/index.tsx:16]: We try to avoid using JS for responsiveness. Why is this needed?

File: src/components/common/PageLayout/index.tsx [hunk 2]
```diff
@@ -1,43 +1,46 @@
...
+
+  const [appUrl] = useSafeAppUrl()
+  const isSafeAppPage = !!appUrl
+  const isShareSafeAppPage = router.pathname === AppRoutes.share.safeApp
 
-  const onMenuToggle = (): void => {
-    setIsMobileDrawerOpen((prev) => !prev)
-  }
+  const [isSideBarCollapsed, setIsSideBarCollapsed] = useState<boolean>(isMobile)
```
Comment (katspaugh) [inline src/components/common/PageLayout/index.tsx:22]: Let's spell sidebar as a single word, it's not a "side bar".

File: src/components/common/PageLayout/styles.module.css [hunk 3]
```diff
@@ -1,18 +1,11 @@
-.header {
-  position: fixed;
-  left: 0;
-  top: 0;
-  width: 100%;
-  z-index: 1201;
-}
```
Comment (katspaugh) [inline src/components/common/PageLayout/styles.module.css:7]: Why move this? The logic behind it was that PageLayout orchestrates the layout, and the header is only responsible itself, not its position on the page.

Comment (katspaugh) [inline src/components/common/PageLayout/styles.module.css:7]: Why move this? PageLayout is responsible for the positions of the elements in it. Header is only responsible for what's inside the header.

File: src/components/safe-apps/AppFrame/TransactionQueueBar/styles.module.css [hunk 4]
```diff
@@ -3,7 +3,7 @@
   bottom: 0;
   right: 0;
   width: 100%;
-  z-index: 1;
+  z-index: 1300;
```
Comment (katspaugh) [inline src/components/safe-apps/AppFrame/TransactionQueueBar/styles.module.css:6]: Please check how this plays together with the onboard modals.

File: src/components/sidebar/Sidebar/index.tsx [hunk 5]
```diff
@@ -14,54 +20,82 @@ import { trackEvent, OVERVIEW_EVENTS } from '@/services/analytics'
 import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
 import OwnedSafes from '../OwnedSafes'
 
-const Sidebar = (): ReactElement => {
+type SidebarProps = {
+  hideSidebar: boolean
```
Comment (katspaugh) [inline src/components/sidebar/Sidebar/index.tsx:24]: `isHidden`?

File: src/components/sidebar/Sidebar/index.tsx [hunk 6]
```diff
@@ -14,54 +20,82 @@ import { trackEvent, OVERVIEW_EVENTS } from '@/services/analytics'
...
+  showCollapseSidebarButton: boolean
+  collapseSideBar: () => void
```
Comment (katspaugh) [inline src/components/sidebar/Sidebar/index.tsx:26]: Is this a callback? If so it should be called `onCollapse`.

**Review by katspaugh — CHANGES_REQUESTED** [review]

File: src/components/safe-apps/AppFrame/TransactionQueueBar/styles.module.css (hunk 4 above)
Comment (DaniSomoza) [reply src/components/safe-apps/AppFrame/TransactionQueueBar/styles.module.css:6]: I will use the `z-index: var(--onboard-modal-z-index, var(--modal-z-index));`   (`z-index: 1201;`) to make sure that the Transaction queue bar does not interfere with the onboard modals:

Captura de pantalla 2022-11-16 a las 15 52 53

File: src/components/sidebar/Sidebar/index.tsx (hunk 5 above)
Comment (DaniSomoza) [reply src/components/sidebar/Sidebar/index.tsx:24]: yep! Maybe something like `isSidebarHidden` ?

File: src/components/common/PageLayout/index.tsx (hunk 1 above)
Comment (DaniSomoza) [reply src/components/common/PageLayout/index.tsx:16]: I think that we need this because we need to set the variant of the Drawer based on the resolution in the Sidebar component:

```
      <Drawer
        variant={isMobile ? 'temporary' : 'persistent'}
```

File: src/components/sidebar/Sidebar/index.tsx (hunk 6 above)
Comment (DaniSomoza) [reply src/components/sidebar/Sidebar/index.tsx]: its the function that sets in the React state of the sidebar is open or closed:

```
  const [isSidebarCollapsed, setIsSidebarCollapsed] = useState<boolean>(isMobile)

  const collapseSidebar = () => setIsSidebarCollapsed((prev) => !prev)
```

File: src/components/sidebar/Sidebar/index.tsx [hunk 7]
```diff
@@ -14,54 +20,82 @@ import { trackEvent, OVERVIEW_EVENTS } from '@/services/analytics'
 import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
 import OwnedSafes from '../OwnedSafes'
 
-const Sidebar = (): ReactElement => {
+type SidebarProps = {
+  hideSidebar: boolean
+  showCollapseSidebarButton: boolean
```
[the remark this answers is not in the document]
Comment (DaniSomoza) [reply src/components/sidebar/Sidebar/index.tsx:25]: we need these 3 props in the Sidebar componet:

- `isSidebarHidden` prop is used by the <Drawer open={!isSidebarHidden} /> component show or hide the sidebar
- `showCollapseSidebarButton` prop is used to show or not the implemented feature
- `collapseSidebar` prop sets in the React state of the sidebar is open or closed

File: src/components/sidebar/Sidebar/styles.module.css [hunk 8]
```diff
@@ -64,11 +65,41 @@
...
+  height: calc(100vh - var(--header-height));
+  width: var(--show-sidebar-bar-width);
+  top: var(--header-height);
+  right: calc(var(--show-sidebar-bar-width) * -1);
+  background-color: var(--color-border-light);
+  transition: background-color 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  cursor: pointer;
+  z-index: 1;
+  clip-path: inset(0 -14px 0 0);
+}
```
[the remark this answers is not in the document]
Comment (DaniSomoza) [reply src/components/sidebar/Sidebar/styles.module.css:80]: I think that there is nothing similar in MUI right? I created the component for this feature using CSS

File: src/components/common/PageLayout/index.tsx (hunk 1 above)
Comment (katspaugh) [reply src/components/common/PageLayout/index.tsx:16]: I mean, before we were rendering two sidebars, for mobile and for desktop, why did you have to change that?

File: src/components/sidebar/Sidebar/index.tsx (hunk 5 above)
Comment (katspaugh) [reply src/components/sidebar/Sidebar/index.tsx]: The whole component is called sidebar, why prefix each prop with the same word?

File: src/components/sidebar/Sidebar/index.tsx (hunk 6 above)
Comment (katspaugh) [reply src/components/sidebar/Sidebar/index.tsx]: Yeah, so it’s a callback.

File: src/components/sidebar/Sidebar/index.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (katspaugh) [reply src/components/sidebar/Sidebar/index.tsx:25]: I understand that, what I mean is we could use the callback in a double role: if passed — shows the button. If not passed don’t show the button.

File: src/components/sidebar/Sidebar/styles.module.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (katspaugh) [reply src/components/sidebar/Sidebar/styles.module.css:80]: The mobile sidebar for example slides in and out from the left side of the screen. It’s a built in animation in MUI.

[the remark this answers is not in the document]
Comment (katspaugh) [reply src/components/sidebar/Sidebar/styles.module.css:80]: See SwipeableDrawer https://mui.com/material-ui/react-drawer/#swipeable

**Review by katspaugh — APPROVED** [review]
Thanks for being so patient!
Really nice work!

**Review by dasanra — APPROVED** [review]

**Comment by katspaugh** [comment]
Please pull from dev so that the `Deploy to dev/staging` job is triggered.

File: src/components/common/PageLayout/SideDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  return pathname === AppRoutes.share.safeApp
+}
+
+const isSafeAppRoute = (pathname: string, query: ParsedUrlQuery): boolean => {
+  return pathname === AppRoutes.apps && !!query.appUrl
+}
+
+const SideDrawer = ({ isOpen, onToggle }: SideDrawerProps): ReactElement => {
+  const { pathname, query } = useRouter()
+  const isSmallScreen = useMediaQuery('(max-width: 900px)')
```
Comment (iamacook) [inline src/components/common/PageLayout/SideDrawer.tsx:28]: Can we use the `md` theme variable here instead? It's also `900px`. You can access it via `useTheme`.

**Review by katspaugh — APPROVED** [review]
🚀

**Review by iamacook — APPROVED** [review]
